compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java

```diff
@@ -322,7 +322,8 @@ public Optional<TypeSymbol> typeOf(LineRange range) {
     }
 
     private BType getDeterminedType(BLangNode node, LineRange range) {
-        if (node.getKind() == NodeKind.INVOCATION && node.getDeterminedType().getKind() == TypeKind.FUTURE) {
+        if (node.getKind() == NodeKind.INVOCATION && node.getDeterminedType() != null
```

Comment:
When can `node.getDeterminedType()` be null?

Reply:
For this invocation node `"id2".post()` we get the determined type as null.